copy.go

```diff
@@ -54,6 +56,40 @@ type CopyOptions struct {
 	MapRoot func(ctx context.Context, src content.Storage, root ocispec.Descriptor) (ocispec.Descriptor, error)
 }
 
+// selectPlatform implements platform filter and returns the descriptor of
+// the first matched manifest from the manifest list / image index.
+func (o *CopyOptions) selectPlatform(ctx context.Context, src content.Storage, root ocispec.Descriptor, p *ocispec.Platform) (ocispec.Descriptor, error) {
+	if root.MediaType != docker.MediaTypeManifestList && root.MediaType != ocispec.MediaTypeImageIndex {
```

Comment:
I'm wondering, what if the root is not a multi-platform manifest?

Reply:
For now, the code checks the MediaType of the node first. If it is a manifest list or image index, then will call func `Successors` to fetch the nodes directly pointed by the current node, and then compare the returned manifests with the target platform. Otherwise we will return `ErrUnsupported` error.

For other media types [OCI Image Media Types](https://github.com/opencontainers/image-spec/blob/main/media-types.md) (such as Image Manifest), since they do not contain the platform property, I think it is reasonable to return the ErrUnsupported error. Besides, [crane](https://github.com/google/go-containerregistry/blob/5749ee68eb46376d695c495d15f251943fd0759f/pkg/v1/remote/descriptor.go#L152) only supports platform filter for image index/manifest list as well.

Do you have any suggestions for the expected behavior for a non multi-platform manifest? 
cc @FeynmanZhou @yizha1

Reply:
Makes sense.

Reply:
For Docker/OCI image manifests, the platform information is stored in the config blob. You may open it and try to inspect the platform info to match the platform.

Reply:
I'm wondering whether it is necessary to support platform selection for a manifest.